esx_service/vmdk_ops.py

```diff
@@ -402,8 +402,15 @@ def vol_info(vol_meta, vol_size_info, datastore):
 # Return error, or None for OK
 def removeVMDK(vmdk_path):
     logging.info("*** removeVMDK: %s", vmdk_path)
-    cmd = "{0} {1}".format(VMDK_DELETE_CMD, vmdk_path)
 
+    # Check the current volume status
+    attached, uuid, attach_as = getStatusAttached(vmdk_path)
+    if attached:
```

Comment:
We cannot use a lock on -flat, because there is no -flat on VSAN or VVOL :-).
I think the workaround with relying on metadata is good enough. We can recover metadata if it's corrupted. We can even automated it with something like 'vmdkops_admin service metadata rebuild' (not in this PR )